src/metabase/api/collection.clj

```diff
@@ -406,14 +406,23 @@
                      :from   [[{:union-all queries} :dummy_alias]]}
         rows-query  {:select   [:*]
                      :from     [[{:union-all queries} :dummy_alias]]
-                     :order-by sql-order
-                     :limit    offset-paging/*limit*
-                     :offset   offset-paging/*offset*}]
-    {:total  (->> (db/query total-query) first :count)
-     :data   (->> (db/query rows-query) post-process-rows)
-     :limit  offset-paging/*limit*
-     :offset offset-paging/*offset*
-     :models models}))
+                     :order-by sql-order}
+        ;; We didn't implement collection pagination for snippets namespace for root/items
+        ;; Rip out the limit for now and put it back in when we want it
+        limit-query (if (= (:collection-namespace options) "snippets")
```

Comment:
Just for the future, let's try to refrain from extra work in a PR to fix a P1. Makes it harder to be secure that we are delivering a fix for the P1 without introducing other errors.